graphstorm-processing/scripts/submit_gsp_emr_step.py

```diff
@@ -0,0 +1,245 @@
+"""
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License").
+You may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Launches an EMR on EC2 cluster a submits a GSProcessing job as an EMR Step.
+
+Usage:
+    python submit_gsp_emr_step.py \
+            --entry-point-s3 ${S3_ENTRY_POINT} \
+            --worker-count ${NUM_WORKERS} \
+            --instance-type ${INSTANCE_TYPE} \
+            --gsp-arguments "--config-filename ${CONFIG_FILE} \
+                --input-prefix ${INPUT_PREFIX} \
+                --output-prefix ${OUTPUT_PREFIX} \
+                --add-reverse-edges ${GENERATE_REVERSE} \
+                --do-repartition ${REPARTITION_ON_LEADER}"
+"""
+
+import argparse
+import os
+from pprint import pprint
+
+import boto3
+
+
+def parse_args():
+    """Parses script arguments."""
+    parser = argparse.ArgumentParser(description="Launch a GSP EMR task")
+
+    # Cluster setup params
+    parser.add_argument(
+        "--gsp-ecr-repository-name",
+        default="graphstorm-processing-emr",
+        help=("GSProcessing EMR on EC2 ECR repository name, default graphstorm-processing-emr"),
+    )
+    parser.add_argument(
+        "--gsp-ecr-registry",
+        required=False,
+        default=None,
+        help=("GSProcessing ECR registry, default <ACCOUNT>.dkr.ecr.<REGION>.amazonaws.com"),
+    )
+    parser.add_argument(
+        "--gsp-image-tag",
+        default="latest-x86_64",
+        help=("GSProcessing EMR image tag, default is latest-x86_64"),
+    )
+    parser.add_argument("--timeout-hours", type=int, default=1)
+    parser.add_argument(
+        "--worker-count",
+        type=int,
+        required=True,
+        help="Number of worker (core) instances. Required",
+    )
+    parser.add_argument(
+        "--instance-type",
+        required=True,
+        help="EC2 instance type to use for the EMR cluster. Required.",
+    )
+    parser.add_argument(
+        "--log-uri",
+        default=None,
+        help="S3 URI to use for log storage. If not provided, no logs are stored.",
+    )
+    parser.add_argument(
+        "--service-role",
+        default="EMR_DefaultRole",
```

Comment:
I do think they now have V2 role here, may we refer to the V2? https://docs.aws.amazon.com/emr/latest/ManagementGuide/emr-iam-role.html